From 33dac936dd839354359e2f0002136749d1e0339e Mon Sep 17 00:00:00 2001
From: Francisco Franco <franciscofranco.1990@gmail.com>
Date: Wed, 13 Nov 2013 01:21:32 +0000
Subject: [PATCH 066/517] Revert "suspend: remove disable/enable cpus every
 time suspend is called (which is called every damn
 5 seconds -.-)."

This reverts commit 450a7acdebdc691937c8a2ed9515bab598358b21.

Signed-off-by: Simarpreet Singh <simar@linux.com>
---
 kernel/power/suspend.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kernel/power/suspend.c b/kernel/power/suspend.c
index 1172f2b..7ed82c8 100644
--- a/kernel/power/suspend.c
+++ b/kernel/power/suspend.c
@@ -198,6 +198,10 @@ static int suspend_enter(suspend_state_t state, bool *wakeup)
 	if (suspend_test(TEST_PLATFORM))
 		goto Platform_wake;
 
+	error = disable_nonboot_cpus();
+	if (error || suspend_test(TEST_CPUS))
+		goto Enable_cpus;
+
 	arch_suspend_disable_irqs();
 	BUG_ON(!irqs_disabled());
 
@@ -214,6 +218,9 @@ static int suspend_enter(suspend_state_t state, bool *wakeup)
 	arch_suspend_enable_irqs();
 	BUG_ON(irqs_disabled());
 
+ Enable_cpus:
+	enable_nonboot_cpus();
+
  Platform_wake:
 	if (need_suspend_ops(state) && suspend_ops->wake)
 		suspend_ops->wake();
-- 
1.7.9.5

